samples/wifi/promiscuous/src/main.c

```diff
@@ -13,6 +13,320 @@ LOG_MODULE_REGISTER(promiscuous, CONFIG_LOG_DEFAULT_LEVEL);
 
 #include "wifi_connection.h"
 #include <zephyr/net/ethernet.h>
+#include <zephyr/posix/unistd.h>
+#include <zephyr/posix/sys/socket.h>
+
+#define STACK_SIZE CONFIG_RX_THREAD_STACK_SIZE
+#define THREAD_PRIORITY K_PRIO_COOP(CONFIG_NUM_COOP_PRIORITIES - 1)
+#define RECV_BUFFER_SIZE 1024
+#define RAW_PKT_HDR 6
+
+/* Enumeration for Wi-Fi frame types and subtypes */
+enum frame_type {
+	MANAGEMENT_FRAME = 0x00,
+	CONTROL_FRAME = 0x01,
+	DATA_FRAME = 0x02,
+	RESERVED_FRAME = 0x03,
+	UNKNOWN_FRAME = 0xFF
+};
+
+enum control_subtype {
+	CF_END_SUBTYPE = 0x0E,
+	ACK_SUBTYPE = 0x0D,
+	RTS_SUBTYPE = 0x0B,
+	CTS_SUBTYPE = 0x0C,
+	BLOCK_ACK_SUBTYPE = 0x09,
+	BLOCK_ACK_REQ_SUBTYPE = 0x08
+};
+
+enum data_subtype {
+	DATA_SUBTYPE = 0x00,
+	QOS_DATA_SUBTYPE = 0x08,
+	NULL_SUBTYPE = 0x04,
+	QOS_NULL_SUBTYPE = 0x0C
+};
+
+enum management_subtype {
+	PROBE_REQ_SUBTYPE = 0x04,
+	BEACON_SUBTYPE = 0x08,
+	PROBE_RESPONSE_SUBTYPE = 0x05,
+	ACTION_SUBTYPE = 0x0D
+	/* Add more subtypes as needed */
+};
+
+struct packet_data {
+	int send_sock;
+	int recv_sock;
+	char recv_buffer[RECV_BUFFER_SIZE];
+};
+
+/* Structure for the MAC header */
+typedef struct {
+	unsigned char frame_control;
+	unsigned char duration[2];
+	unsigned char receiver_addr[6];
+	unsigned char transmitter_addr[6];
+	/* Add more fields based on your actual MAC header structure */
```

Comment:
> your actual MAC header

isn't this defined by protocol?

Reply:
Added a separate structure in sample to handle.